pkg/operator/injector/injector.go

```diff
@@ -0,0 +1,487 @@
+	defaultConfigmapNamespace   = "skywalking-swck-system"
+	agentConfigAnnotationPrefix = "agent.skywalking.apache.org/"
+	// If user want to use other Plugins' config ,the annotation must have the following form
+	// plugins.skywalking.apache.org/${config.name} = ${config.value}
+	// for example , if user want to enable plugin.mongodb.trace_param
+	// the annotation is plugins.skywalking.apache.org/plugin.mongodb.trace_param: "true"
+	otherPluginsAnnotationPrefix = "plugins.skywalking.apache.org/"
+)
+
+// log is for logging in this package.
+var log = logf.Log.WithName("injector")
+
+// SidecarInjectField contains all info that will be injected
+type SidecarInjectField struct {
+	// determine whether to inject , default is not to inject
+	needInject bool
+	// determine whether to use annotation to overlay agent config ,
+	// default is not to overlay，which means only use configmap to set agent config
+	// Otherwise, the way to overlay is set jvm agent ,just like following
+	// -javaagent: /sky/agent/skywalking-agent,jar={config1}={value1},{config2}={value2}
+	agentOverlay bool
+	// initcontainer is a container that has the agent folder
+	initcontainer corev1.Container
+	// sidecarVolume is a shared directory between App's container and initcontainer
+	sidecarVolume corev1.Volume
+	// sidecarVolumeMount is a path that specifies a shared directory
+	sidecarVolumeMount corev1.VolumeMount
+	// configmapVolume is volume that provide user with configmap
+	configmapVolume corev1.Volume
+	// configmapVolumeMount is the configmap's mountpath for user
+	// Notice : the mount path will overwrite the original agent/config/agent.config
+	// So the mount path must match the path of agent.config in the image
+	configmapVolumeMount corev1.VolumeMount
+	// env is used to set java agent’s parameters
+	env corev1.EnvVar
+	// the string is used to set jvm agent ,just like following
+	// -javaagent: /sky/agent/skywalking-agent,jar=jvmAgentConfigStr
+	jvmAgentConfigStr string
+	// determine which container to inject , default is to inject all containers
+	injectContainer string
+}
+
+// NewSidecarInjectField will create a new SidecarInjectField
+func NewSidecarInjectField() *SidecarInjectField {
+	return new(SidecarInjectField)
+}
+
+// Inject will do real injection
+func (s *SidecarInjectField) Inject(pod *corev1.Pod) {
+	// add initcontrainers to spec
+	if pod.Spec.InitContainers != nil {
+		pod.Spec.InitContainers = append(pod.Spec.InitContainers, s.initcontainer)
+	} else {
+		pod.Spec.InitContainers = []corev1.Container{s.initcontainer}
+	}
+
+	// add volume to spec
+	if pod.Spec.Volumes != nil {
+		pod.Spec.Volumes = append(pod.Spec.Volumes, s.sidecarVolume, s.configmapVolume)
+	} else {
+		pod.Spec.Volumes = []corev1.Volume{s.sidecarVolume, s.configmapVolume}
+	}
+
+	// add volumemount and env to container
+	for i, c := range pod.Spec.Containers {
+		// choose a specific container to inject
+		if s.injectContainer != "" && c.Name == s.injectContainer {
```

Comment:
Supposing the first container is not the `injectContainer`, does it get injected? If true, it should be a bug.

Reply:
Yes,it's a bug, if injectContainer is set , the container before injectContainer in Spec will be injected. 
Thanks for correcting.